public/app/features/profile/ChangePasswordForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { PureComponent, MouseEvent } from 'react';
+import config from 'app/core/config';
+import { Button, LinkButton, LoadingPlaceholder } from '@grafana/ui';
+import { ChangePasswordFields } from 'app/core/utils/UserProvider';
+import { PasswordInput } from 'app/core/components/PasswordInput/PasswordInput';
+
+export interface Props {
+  isLoading: boolean;
+  onChangePassword: (payload: ChangePasswordFields) => void;
+}
+
+export interface State {
+  oldPassword: string;
+  newPassword: string;
+  confirmNew: string;
+}
+
+export class ChangePasswordForm extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      oldPassword: '',
+      newPassword: '',
+      confirmNew: '',
+    };
+  }
+
+  onOldPasswordChange(oldPassword: string) {
+    this.setState({ oldPassword });
+  }
+
+  onNewPasswordChange(newPassword: string) {
+    this.setState({ newPassword });
+  }
+
+  onConfirmPasswordChange(confirmNew: string) {
+    this.setState({ confirmNew });
+  }
+
+  render() {
+    const { oldPassword, newPassword, confirmNew } = this.state;
+    const { onChangePassword, isLoading } = this.props;
+    const { ldapEnabled, authProxyEnabled } = config;
+
+    if (isLoading) {
```

Comment:
The naming here feels very confusing, what is loading, in this case it's the action taking place and nothing is really loading, so maybe isChanging or isSaving? and the whole form is hidden and replaced with text feels a bit wrong. The action button should change state to disabled and show spinner instead. @dprokop maybe long term this should be a built-in feature in the button or what do you think?

Reply:
It definitely should, ideally we could have `AsyncState` type that multiple components could accept (i.e. Button) and reflect it in the UI.

Reply:
I agree with @torkelo regarding the `isLoading` naming. Since ChangePasswordForm is a component that reacts to async state, I think it should accept sth like `AsyncState` type prop thhat would come from the async actions provider, in this case its `UserProvider`. 

So for this particular case the user provider could expose an object that represents the states of different async actions:

```
{
  changePassword: AsyncState;
}
```

where AsyncState could be sth like:
```
interface AsyncState  {
  state: 'pending' | 'inProgress' | 'error' | 'success';
  data: any; // To i.e. handle information about error that ocured
}
```